src/InvariantTheory/iterators.jl

```diff
@@ -1,23 +1,85 @@
-
 ################################################################################
 #
 #  All Monomials
 #
 ################################################################################
 
-# Return an iterator over all monomials of R of degree d
-all_monomials(R::MPolyRing, d::Int) = AllMonomials(R, d)
+@doc raw"""
+    monomials_of_degree(R::MPolyRing, d::Int)
+    monomials_of_degree(R::MPolyRing, d::Int, vars::Vector{Int})
+    monomials_of_degree(R::MPolyRing, d::Int, r::UnitRange{Int})
+
+Return an iterator over all monomials in `R` of degree `d`.
+An additional vector or range of integers may be given to specify a subset of
+variables of `R` to be used.
+
+# Examples
+```jldoctest
+julia> R, (x, y, z) = QQ["x", "y", "z"];
+
+julia> collect(monomials_of_degree(R, 3))
+10-element Vector{QQMPolyRingElem}:
+ x^3
+ x^2*y
+ x^2*z
+ x*y^2
+ x*y*z
+ x*z^2
+ y^3
+ y^2*z
+ y*z^2
+ z^3
+
+julia> sum(monomials_of_degree(R, 4, 1:2))
+x^4 + x^3*y + x^2*y^2 + x*y^3 + y^4
+
+julia> sum(monomials_of_degree(R, 4, [1, 3]))
+x^4 + x^3*z + x^2*z^2 + x*z^3 + z^4
+```
+"""
```

Comment:
```suggestion
"""
monomials_of_degree(

```

Reply:
Then the docstring is attached to the function and not one particular method.

Reply:
Thanks, I did not know about this.